### PR TITLE
refactor: update cache key and remove session dependency

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-analytics-data-api doesn't install it
    - `test-master.in` if edx-analytics-data-api pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the edx-analytics-data-api requirements installed or if you checked out edx-enterprise-data into the src directory used by edx-analytics-data-api, you can run this command through an edx-analytics-data-api shell.
        - It would be `./manage.py makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise-data/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise-data/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/edx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_data/management/commands/tests/test_create_enterprise_learner_enrollment_lpr_v1.py [hunk 1]
```diff
@@ -57,13 +57,13 @@ def test_create_enterprise_learner_enrollment_lpr_v1_with_dsc_enabled(self):
...
-        assert enterprise_learner_enrollment[0].user_username is None
-        assert enterprise_learner_enrollment[0].enterprise_user is None
-        assert enterprise_learner_enrollment[0].user_email is None
+        assert enterprise_learner_enrollment[0].letter_grade is not None
+        assert enterprise_learner_enrollment[0].last_activity_date is not None
+        assert enterprise_learner_enrollment[0].progress_status is not None
+        assert enterprise_learner_enrollment[0].enterprise_user_id is not None
+        assert enterprise_learner_enrollment[0].user_username is not None
+        assert enterprise_learner_enrollment[0].enterprise_user is not None
+        assert enterprise_learner_enrollment[0].user_email is not None
```
Comment (HammadAhmadWaqas) [inline enterprise_data/management/commands/tests/test_create_enterprise_learner_enrollment_lpr_v1.py:66]: Why do we have to revert the checks for this test and the test above?

**Review by HammadAhmadWaqas — APPROVED** [review]
Just 1 comment, Otherwise looks good.

File: enterprise_data/management/commands/tests/test_create_enterprise_learner_enrollment_lpr_v1.py (hunk 1 above)
Comment (muhammad-ammar) [reply enterprise_data/management/commands/tests/test_create_enterprise_learner_enrollment_lpr_v1.py:66]: Due to https://github.com/openedx/edx-enterprise-data/pull/341/files#diff-6cc15faa047fb8c27baf0f209d3a39628c40fd0be410729e877d815494681d30R231

File: enterprise_data/tests/test_utils.py [hunk 2]
```diff
@@ -229,7 +228,7 @@ def set_fields_according_to_consent(
             'user_country_code',
             'user_username',
         ]
-        if create and obj.is_consent_granted:
+        if create and not obj.is_consent_granted:
```
Comment (HammadAhmadWaqas) [inline enterprise_data/tests/test_utils.py:231]: So there was a bug in the existing logic?

Comment (muhammad-ammar) [reply enterprise_data/tests/test_utils.py:231]: yes.

Comment (muhammad-ammar) [reply enterprise_data/tests/test_utils.py:231]: The logic is to set couple of fields to `None` if `is_consent_granted` is `False`. But the above logic in the factory class is doing the opposite.

**Pull request merged**